src/datatypes/array.c

```diff
@@ -35,6 +35,22 @@ uint32_t SIArray_Length(SIValue siarray) {
 	return array_len(siarray.array);
 }
 
+bool SIArray_ContainsType(SIValue siarray, SIType t) {
+	uint array_len = SIArray_Length(siarray);
+	for(uint i = 0; i < array_len; i++) {
+		SIValue elem = SIArray_Get(siarray, i);
+		if(SI_TYPE(elem) & t) return true;
+
+		// recursively check nested arrays
+		if(SI_TYPE(elem) == T_ARRAY) {
```

Comment:
For completeness, I think we should also recursively test `Map` data type
add unit-test for this function as part of `test_value.cpp` or introduce `test_array.cpp`

Reply:
I don't quite follow the first part of this request. `Map` types cannot be used as property values, so the logic in this PR disallows them in the same way that a node or edge value would be disallowed.

Reply:
Although you're looking at this from a specific context.